apps/server/src/modules/board/service/submission-item.service.ts

```diff
@@ -43,17 +45,35 @@ export class SubmissionItemService {
 	}
 
 	async update(submissionItem: SubmissionItem, completed: boolean): Promise<void> {
+		const parent = await this.getParent(submissionItem);
+
+		submissionItem.completed = completed;
+
+		await this.boardDoRepo.save(submissionItem, parent);
+	}
+
+	async delete(submissionItem: SubmissionItem): Promise<void> {
+		await this.getParent(submissionItem);
+
+		await this.boardDoRepo.delete(submissionItem);
+	}
+
+	private async getParent(submissionItem: SubmissionItem): Promise<SubmissionContainerElement> {
 		const submissionContainterElement = await this.boardDoRepo.findParentOfId(submissionItem.id);
+
 		if (!isSubmissionContainerElement(submissionContainterElement)) {
 			throw new UnprocessableEntityException();
 		}
 
+		this.checkNotLocked(submissionContainterElement);
```

Comment:
rename to `checkDueDate`

It is not very clear to me if we set the dueDate on the beginning or the end of the day.
I know the UI has a time picker, but what if it is not set?

Reply:
the logic is older: If due date is set from UI by the teacher, and is already expired, students cannot check "completed" anymore. 
<img width="867" alt="Bildschirmfoto 2024-02-06 um 16 29 14" src="https://github.com/hpi-schul-cloud/schulcloud-server/assets/17074330/f4242756-1a19-4313-acae-ba0905ce96d3">